ui/job-view/pushes/Platform.jsx

```diff
@@ -16,41 +19,105 @@ PlatformName.propTypes = {
   title: PropTypes.string.isRequired,
 };
 
-export default function Platform(props) {
-  const {
-    platform,
-    repoName,
-    filterPlatformCb,
-    filterModel,
-    pushGroupState,
-    duplicateJobsVisible,
-    groupCountsExpanded,
-  } = props;
-  const { title, groups, id } = platform;
+export default class Platform extends React.PureComponent {
+  static getDerivedStateFromProps(nextProps) {
```

Comment:
A downside to using `getDerivedStateFromProps` is that it will be triggered with any render (like with local state changes) not just prop changes. I wonder if using `componentDidUpdate` here instead would make a difference?

Reply:
Yeah, I was thinking about that, too.  I was hesitant to do a``setState`` in ``componentDidUpdate`` but now that I read up about that more, I guess it's OK.  And that would let me use a ``shouldComponentUpdate`` there which could help a lot.  So, yeah, you're totally right.  This will be the way to go.  Thanks.  :)

Reply:
I should say, doing ``setState`` is fine as long as it's not unchecked...

Reply:
Yeah, I wish the lint rule was smart enough to catch when conditionals are used in that lifecycle.